src/main/java/ru/yandex/practicum/filmorate/service/FilmService.java

```diff
@@ -1,41 +1,169 @@
+        setAllPar(film, mpaList, genres, filmGenres, likes);
+        return film;
+    }
+
+    public void addLike(int userId, int filmId) {
+        Likes like = likesStorage.getLikesCurrentUserWithFilmId(userId, filmId);
+        if (like == null) {
+            likesStorage.addLike(new Likes(filmId, userId));
+        }
     }
 
-    public Film addLikeFilms(int filmId, int userId) {
-        return filmStorage.addLikeFilms(filmId, userId);
+    public void removeLike(int userId, int filmId) {
+        Likes like = likesStorage.getLikesCurrentUserWithFilmId(userId, filmId);
+        if (like == null) {
+            throw new FilmNotFoundException("лайк не найден");
+        }
+        likesStorage.removeLike(new Likes(filmId, userId));
     }
 
     public List<Film> getBestFilms(int count) {
-        return filmStorage.getBestFilms(count);
+        var films = findAllFilms();
+        return films.stream().sorted(Comparator.comparingInt(film -> film.getLikes().size()))
+                .limit(count)
+                .collect(Collectors.toList());
+    }
+
+    public Genres getGenres(int id) {
+        return genreStorage.findGenreById(id);
+    }
+
+    public List<Genres> getGenresList() {
+        return genreStorage.getAllGenres();
+    }
+
+    public Mpa getMpa(int id) {
+        return mpaStorage.getMpaById(id);
+    }
+
+    public List<Mpa> getMpaList() {
+        var mpaList = mpaStorage.getAllMpa();
+        List<Mpa> list = new ArrayList<>(mpaList);
+        list.sort(Comparator.comparing(Mpa::getId));
+        return list;
+    }
+
+    private void addGenreForFilm(Film film) {
+        if (film.getGenres() != null && film.getGenres().size() > 0) {
+            List<Genres> ratingList = film.getGenres();
+            Set<Integer> uniqueId = new HashSet<>();
+            ratingList.removeIf(rating -> !uniqueId.add(rating.getId()));
+            film.setGenres(ratingList);
+            for (var genre : film.getGenres()) {
+                var filmGenre = new FilmGenre(film.getId(), genre.getId());
+                filmGenreStorage.add(filmGenre);
+            }
+        }
+    }
+
+    private void setAllPar(Film film, Set<Mpa> mpaList, List<Genres> genres,
+                           List<FilmGenre> filmGenres, Set<Likes> likes) {
+        List<Genres> genreByFilm = new ArrayList<>();
+        filmGenres.stream().filter(f -> f.getFilmId() == film.getId())
+                .forEach(f -> genreByFilm.add(new Genres(f.getGenreId(),
+                                genres.stream().filter(g -> g.getId() == f.getGenreId()).findAny().get().getName())));
+
+        film.setGenres(genreByFilm);
+        film.getMpa().setName(mpaList.stream().filter(m -> m.getId() == film.getMpa().getId()).findAny().get().getName());
+        Set<Integer> likesByFilm = new HashSet<>();
+        likes.stream().filter(l -> l.getFilmId() == film.getId()).forEach(l -> likesByFilm.add(l.getUserId()));

```

Comment:
Стримы принято бить по строкам, чтобы улучшить читаемости кода
```suggestion
        likes.stream()
            .filter(l -> l.getFilmId() == film.getId())
            .forEach(l -> likesByFilm.add(l.getUserId()));
```

Поправь для всех стримов в проекте